torchrl/data/tensor_specs.py

```diff
@@ -3678,6 +3679,99 @@ def clone(self) -> Categorical:
         )
 
 
+class Choice(TensorSpec):
+    """A discrete choice spec for either tensor or non-tensor data.
+
+    Args:
+        stack (:class:`~Stacked`, :class:`~StackedComposite`, or :class:`~tensordict.NonTensorStack`):
+            Stack of specs or non-tensor data from which to choose during
+            sampling.
```

Comment:
See test comment: I think we should be able to get away with just a list of specs